File: app/src/main/java/com/dani_chuks/andeladeveloper/starwars/data/db/IntegerListConverter.java
package com.dani_chuks.andeladeveloper.starwars.data.db;

import android.arch.persistence.room.TypeConverter;

import java.util.ArrayList;
import java.util.List;

public class IntegerListConverter {
    @TypeConverter
    public List<Integer> gettingListFromString(String urlIds) {
        List<Integer> list = new ArrayList<>();

        String[] array = urlIds.split(",");

        for (String s : array) {
            list.add(Integer.parseInt(s));
        }
        return list;
    }

    @TypeConverter
    public String writingStringFromList(List<Integer> list) {
        String urlId = "";
        for (int i : list) {
            urlId += "," + i;
        }
        return urlId;
    }
}

File: app/src/main/java/com/dani_chuks/andeladeveloper/starwars/data/db/IntegerListConverter.kt
package com.dani_chuks.andeladeveloper.starwars.data.db

import androidx.room.TypeConverter
import java.lang.StringBuilder

import java.util.ArrayList

class IntegerListConverter {
    @TypeConverter
    fun gettingListFromString(urlIds: String): List<Int> =
            urlIds.split(",".toRegex())
                    .dropLastWhile { it.isEmpty() }
                    .toTypedArray()
                    .map { Integer.parseInt(it) }
                    .toList()

    @TypeConverter
    fun writingStringFromList(list: List<Int>): String {
        val stringBuilder = StringBuilder()
        list.forEach { 
            stringBuilder.append(",$it")
        }
        return stringBuilder.toString()
    }
}
